CMS-Enterprise/easi-app
# [EASI-4330] GRB reviewer admin view

# EASI-4330

## Description


- Created conditional rendering on the admin view for GRB reviewers
  - ~~useAuth hook with helper functions to return user job roles~~
  - `IsGrbReviewer` context to render GRB reviewer view within admin pages
    - Provider in [GovernanceReviewTeam](https://github.com/CMS-Enterprise/easi-app/blob/fe847b7c82eb457fa41dbdbbb9504f4b4463a8f8/src/views/GovernanceReviewTeam/index.tsx#L30-L33) and context usage example [here](https://github.com/CMS-Enterprise/easi-app/blob/fe847b7c82eb457fa41dbdbbb9504f4b4463a8f8/src/views/GovernanceReviewTeam/IntakeReview/index.tsx#L20)
- Updated subpages, summary header, and navigation to match [Figma](https://www.figma.com/design/G0UN35V7SY0g9nkTIESxsY/EASi-GRB-Experience?node-id=2226-3094)
  - Updated [subNavItems array](https://github.com/CMS-Enterprise/easi-app/blob/EASI-4330/grb-reviewer-admin-view/src/views/GovernanceReviewTeam/subNavItems.ts) to return correct navigation items based on admin view
- Created new route for GRB reviewers (/governance-review-board/:id/)
  - Updated routing so that GRB reviewers cannot access GRT view


## How to test this change

1. Check that GRB reviewer view displays "page cannot be found" if user is NOT a GRB reviewer (/governance-review-board/a2fa0d4b-909f-45d8-ad8c-90f22cf0db19/intake-request)
2. Use gql playground to add user as GRB reviewer to system intake
```graphql
mutation createGrbReviewer {
  createSystemIntakeGRBReviewer(input:{
    systemIntakeID: "a2fa0d4b-909f-45d8-ad8c-90f22cf0db19",
    euaUserId: "ABCD",
    votingRole: VOTING,
    grbRole: CO_CHAIR_CIO
  }){
    id,
    userAccount {
      id
      username
    }
  }
}
```
3. Log in as `ABCD` with only `EASI_P_USER` job code
4. Check that GRB reviewer view renders (/governance-review-board/a2fa0d4b-909f-45d8-ad8c-90f22cf0db19/intake-request)
5. Check that GRT view displays "page cannot be found" (/governance-review-team/a2fa0d4b-909f-45d8-ad8c-90f22cf0db19/intake-request)
6. Log in as `ABCD` with `EASI_D_GOVTEAM` job code
7. Check that GRT view renders (/governance-review-team/a2fa0d4b-909f-45d8-ad8c-90f22cf0db19/intake-request)

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
